applications/tari_app_utilities/src/network_check.rs

```diff
@@ -0,0 +1,70 @@
+//  Copyright 2023. The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use tari_common::{
+    configuration::Network,
+    exit_codes::{ExitCode, ExitError},
+};
+use tari_features::resolver::Target;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum NetworkCheckError {
+    #[error("The network {0} is invalid for this binary built for MainNet")]
+    MainNetBinary(Network),
+    #[error("The network {0} is invalid for this binary built for NextNet")]
+    NextNetBinary(Network),
+    #[error("The network {0} is invalid for this binary built for TestNet")]
+    TestNetBinary(Network),
+}
+
+impl From<NetworkCheckError> for ExitError {
+    fn from(err: NetworkCheckError) -> Self {
+        Self::new(ExitCode::NetworkError, err)
+    }
+}
+
+#[cfg(tari_network_mainnet)]
```

Comment:
Can we add a default to this?
So if the user does not specify a network, it defaults to mainnet

Reply:
There is a default defined as the Target::TestNet. Anytime we do compiled compilation we have to ensure something will be present. For all the feature gating TestNet is the default compilation path.

`#[cfg(all(not(tari_network_mainnet), not(tari_network_nextnet)))]`